zone/guild_mgr.cpp

```diff
@@ -1454,3 +1694,244 @@ bool GuildBankManager::AllowedToWithdraw(uint32 GuildID, uint16 Area, uint16 Slo
+		}
+		else {
+			e->zone_id = 0;				//If zone_id is 0 and we rely on the current world routine, the notes/tribute tabs are not updated for online characters.
+		}
+
+#undef SlideStructString
+#undef PutFieldN
+
+		delete* cur;
+
+		e++;
+	}
+
+	return(retbuffer);
+}
+
+void ZoneGuildManager::SendToWorldMemberLevelUpdate(uint32 guild_id, uint32 level, std::string player_name)
+{
+	auto s_outapp = new ServerPacket(ServerOP_GuildMemberLevelUpdate, sizeof(ServerOP_GuildMessage_Struct));
+	ServerOP_GuildMessage_Struct* s_out = (ServerOP_GuildMessage_Struct*)s_outapp->pBuffer;
+	s_out->guild_id = guild_id;
+	s_out->player_level = level;
+	strn0cpy(s_out->player_name, player_name.c_str(), sizeof(s_out->player_name));
+	worldserver.SendPacket(s_outapp);
+	safe_delete(s_outapp);
+}
+
+void ZoneGuildManager::SendToWorldMemberPublicNote(uint32 guild_id, std::string player_name, std::string public_note)
+{
+	auto s_outapp = new ServerPacket(ServerOP_GuildMemberPublicNote, sizeof(ServerOP_GuildMessage_Struct));
+	ServerOP_GuildMessage_Struct* s_out = (ServerOP_GuildMessage_Struct*)s_outapp->pBuffer;
+	s_out->guild_id = guild_id;
+	strn0cpy(s_out->player_name, player_name.c_str(), sizeof(s_out->player_name));
+	strn0cpy(s_out->note, public_note.c_str(), sizeof(s_out->note));
+	worldserver.SendPacket(s_outapp);
+	safe_delete(s_outapp);
+}
+
+void ZoneGuildManager::SendToWorldMemberRemove(uint32 guild_id, std::string player_name)
+{
+	auto s_outapp = new ServerPacket(ServerOP_GuildMemberRemove, sizeof(ServerOP_GuildMessage_Struct));
+	ServerOP_GuildMessage_Struct* s_out = (ServerOP_GuildMessage_Struct*)s_outapp->pBuffer;
+	s_out->guild_id = guild_id;
+	strn0cpy(s_out->player_name, player_name.c_str(), sizeof(s_out->player_name));
+	worldserver.SendPacket(s_outapp);
+	safe_delete(s_outapp);
+}
+
+void ZoneGuildManager::SendToWorldMemberAdd(uint32 guild_id, uint32 char_id, uint32 level, uint32 _class, uint32 rank, uint32 zone_id, std::string player_name)
+{
+	auto s_outapp = new ServerPacket(ServerOP_GuildMemberAdd, sizeof(ServerOP_GuildMessage_Struct));
+	ServerOP_GuildMessage_Struct* s_out = (ServerOP_GuildMessage_Struct*)s_outapp->pBuffer;
+	s_out->guild_id = guild_id;
+	s_out->player_level = level;
+	s_out->player_rank = rank;
+	s_out->player_zone_id = zone_id;
+	s_out->player_class = _class;
+	strn0cpy(s_out->player_name, player_name.c_str(), sizeof(s_out->player_name));
+	worldserver.SendPacket(s_outapp);
+	safe_delete(s_outapp);
+}
+
+bool ZoneGuildManager::RemoveMember(uint32 guild_id, uint32 char_id, std::string player_name)
+{
+	GuildMembersRepository::DeleteOne(*m_db, char_id);
+	SendToWorldMemberRemove(guild_id, player_name);
+	return true;
+}
+
+void ZoneGuildManager::AddMember(uint32 guild_id, uint32 char_id, uint32 level, uint32 _class, uint32 rank, uint32 zone_id, std::string player_name)
+{
+	BaseGuildMembersRepository::GuildMembers member;
```

Comment:
You can make this a single letter to condense verbosity. `member` is a complex type so `m` doesn't make it any less readable, infact makes it more readable

Reply:
Updated for next commit